acs/repeated_mnist.py

```diff
@@ -21,7 +21,7 @@ def create_repeated_MNIST_dataset(*, num_repetitions: int = 3, add_noise: bool =
         Y_train = torch.concat([Y_train] * num_repetitions)
 
     if add_noise:
-        dataset_noise = torch.empty((len(X_train), 28, 28), dtype=torch.float32).normal_(0.0, 0.1)
+        dataset_noise = torch.empty((len(X_train), 28, 28), dtype=torch.float32).normal_(0.0, 0.1, generator=torch.manual_seed(seed))
```

Comment:
Please replace with

`=torch.Generator().manual_seed(seed)`

Otherwise, the version without `.Generator()` resets the global seed.